t/24-worker-overall.t

```diff
@@ -496,6 +496,20 @@ subtest 'checking and cleaning pool directory' => sub {
     ok(!-f $pid_file, 'PID file deleted when QEMU not running');
 };
 
+subtest 'checking worker address' => sub {
+    my $fqdn_lookup_mock = Test::MockModule->new('OpenQA::Worker::Settings');
+    $fqdn_lookup_mock->redefine(hostfqdn => 'foo');    # only a "short" hostname, not an FQDN
+    $global_settings->{WORKER_HOSTNAME} = 'foo';
+    is $worker->check_availability, undef, 'no error if worker address explicitly specified (also if no FQDN)';
+    $global_settings->{WORKER_HOSTNAME} = undef;    # assume WORKER_HOSTNAME has not been explicitly specified
+    like $worker->check_availability, qr/Unable.*worker address/, 'error if no FQDN detected';
+    $fqdn_lookup_mock->redefine(hostfqdn => 'foo.bar');
+    is $worker->check_availability, undef, 'no error if worker address can be auto-detected';
```

Comment:
I think this doesn't cover the initial case where the shortname is used? Since `check_availability` never specifies a fallback.